app/views/users/_edit_form.html.erb

```diff
@@ -20,14 +20,14 @@
 
       <div class="form-group col-md-6">
         <label for="username"><%= t('users._form.username') %></label>
-        <%= f.text_field :username, { readonly: true, tabindex: 1, placeholder: "Username", class: 'form-control', id: 'username' , title: 'Username field cannot be changed', disabled: true} %>
+        <%= f.text_field :username, { readonly: true, tabindex: 1, placeholder: "Username", class: 'form-control', id: 'username' , title: 'Username field cannot be changed', disabled: true, name: 'username'} %>
         <input type="hidden" name="id" value="<%= @user.username %>" />
       </div>
 
       <div class="form-group col-md-6">
 
         <label for="email"><%= t('users._form.email') %></label>
-        <%= f.text_field :email, { tabindex: 3, required: true, pattern: '[^@]+@[^@]+\.[a-zA-Z]{2,6}', placeholder: "you@email.com", class: 'form-control', id: 'email' } %>
+        <%= f.text_field :email, { tabindex: 3, required: true, pattern: '[^@]+@[^@]+\.[a-zA-Z]{2,6}', placeholder: "you@email.com", class: 'form-control', id: 'email' ,name: 'email'} %>
```

Comment:
I guess the `pattern` here is used to check if the email entered is in correct format or not on the server side. Since the pattern is defined for the server, using the same pattern for email validation on the frontend side js will be good too. Although the email identifier from the jquery-validation library will work for sure but there can be a mismatch in the pattern used by the library and the patterned defined here.

The jquery validation plugin uses the following regex for the email match.
`/^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/`
https://github.com/jquery-validation/jquery-validation/blob/master/src/core.js#L1398

@gauravano @jywarren Please provide your review on what needs to be done for the email validation here.
We can have the following possibilities:
1. The current implementation is good to go :tada: .
2. Use server side pattern in jquery validation. :robot: 
3. Change server side pattern according to the pattern used by validation library. :computer:

Other than this, the PR looks good to me and ready to be merged.

Reply:
Hi @Anupam-dagar Thanks for mentioning this. I forgot to mention this. In the last PR for the same which has been closed and thus I created a new PR.
* I mentioned this too that this existing pattern can't check emails like `lekhika@lekhika` which is valid email 
https://stackoverflow.com/questions/20573488/why-does-html5-form-validation-allow-emails-without-a-dot
* It doesn't allow the lekhika@hotmail.engineer because the regex limited the length of domain name to max 6 characters which isn't correct..

Reply:
I guess then it's good to get the regex changed to the library one since it can accept wide range of emails.

Reply:
Hi @Anupam-dagar I've removed the `hard-coded regex` present in code and replaced it with the new one present in library. It would let us check more conditions which hard-coded was not able to do so.